src/sentry/utils/snuba_rpc.py

```diff
@@ -61,18 +64,46 @@ def meta(
     ): ...
 
 
-def table_rpc(req: TraceItemTableRequest) -> TraceItemTableResponse:
-    resp = _make_rpc_request("EndpointTraceItemTable", "v1", req.meta.referrer, req)
-    response = TraceItemTableResponse()
-    response.ParseFromString(resp.data)
-    return response
+def table_rpc(requests: list[TraceItemTableRequest]) -> list[TraceItemTableResponse]:
+    return make_rpc_requests(table_requests=requests)[0]
 
 
-def timeseries_rpc(req: TimeSeriesRequest) -> TimeSeriesResponse:
-    resp = _make_rpc_request("EndpointTimeSeries", "v1", req.meta.referrer, req)
-    response = TimeSeriesResponse()
-    response.ParseFromString(resp.data)
-    return response
+def timeseries_rpc(requests: list[TimeSeriesRequest]) -> list[TimeSeriesResponse]:
+    return make_rpc_requests(timeseries_requests=requests)[1]
+
+
+def make_rpc_requests(
+    table_requests: list[TraceItemTableRequest] | None = None,
+    timeseries_requests: list[TimeSeriesRequest] | None = None,
+) -> tuple[list[TraceItemTableResponse], list[TimeSeriesResponse]]:
```

Comment:
This method is quite convoluted so let's make this a private API cuz there's likely common uses of making multiples of different RPCs at the same time anyways.

Reply:
@Zylphrex updated